test/org/apache/catalina/connector/TestCoyoteAdapter.java

```diff
@@ -344,6 +345,29 @@ private void doTestNormalize(String input, String expected) {
         }
     }
 
+    @Test
+    public void testCheckNormalize() {
+        doTestCheckNormalize("/url", true);
+
+        doTestCheckNormalize("", false);
+        doTestCheckNormalize("..", false);
+        doTestCheckNormalize("/.", false);
+        doTestCheckNormalize("/..", false);
+        doTestCheckNormalize("/./", false);
+        doTestCheckNormalize("//", false);
+        doTestCheckNormalize("/../", false);
+        doTestCheckNormalize("\\", false);
+        doTestCheckNormalize("\0", false);
+    }
```

Comment:
One test per method here as well. `testCheckNormalize01()`, `testCheckNormalize02()` etc.

While I am not against direct testing of any value that could be passed to `checkNormalize()`, I think the test would be more useful if the appropriate input was passed to `normalize()`, `convertURI()` and then `checkNormalize()`. I appreciate `checkNormalize()` is a public method and could - in theory - receive any input but in this instance the testing needs to focus on what is possible given Tomcat's usage of the method.

Reply:
I split the tests into different functions and removed a few of them since we want to focus the `checkNormalize()` tests to what is possible given its usage.

I did not update the tests to go through `normalize` -> `convertURI` -> `checkNormalize`, but I'd be happy to do this if you think that'd be more appropriate here.